lib/core/src/lib/form/components/form-field/form-field.component.ts

```diff
@@ -88,14 +92,29 @@ export class FormFieldComponent implements OnInit, OnDestroy {
                     instance.fieldChanged.subscribe((field) => {
                         if (field && this.field.form) {
                             this.visibilityService.refreshVisibility(field.form);
-                            field.form.onFormFieldChanged(field);
+                            this.triggerFormFieldChanged(field);
                         }
                     });
+
+                    if (FormFieldTypes.isReactiveType(instance?.field?.type)) {
+                        this.updateReactiveFormControlOnFormRulesEvent(instance);
+                    }
                 }
             }
         }
     }
 
+    private updateReactiveFormControlOnFormRulesEvent(instance: any): void {
+        instance?.formService.formRulesEvent.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(() => {
+            instance?.updateReactiveFormControl();
```

Comment:
Can we add an interface to `instance`? what I mean is adding something like: 

```
interface ReactiveFormWidget {
   updateReactiveFormControl: (): void;
   formService: Type
}
```

and implement it in reactive widgets

```
export class DateTimeWidgetComponent extends WidgetComponent implements OnInit, ReactiveFormWidget {..
```

So `updateReactiveFormControl` could be easily traced?

Reply:
good point, will add it

Reply:
If you add an interface (or an abstract class like WidgetComponent) please double check with a custom form widget if it's all good